docs/guide/headless-cms.md

```diff
@@ -0,0 +1,177 @@
+# Using Apostrophe as a headless CMS
+
+Apostrophe is fully featured as a traditional CMS (with some non-traditional features). It is also immediately ready for use as a headless CMS. There are many scenarios where Apostrophe works well as a headless CMS, including:
+
+- We may want to use Apostrophe to serve webpages normally, but also use the website data in another context, such as a mobile app.
+- We may want to go fully headless with a front end like Gatsby or Gridsome while also allowing editors to use Apostrophe's in-context editing.
+- We may have an app that handles features such as e-commerce but need a CMS to create and pipe content into the app.
+
+Regardless of the reason or the layers involved there are some common concepts to understand.
+
+## Accessing data using the REST APIs
+
+Before going further, keep in mind that there is an [API route reference section](/reference/api/) that will include much of the following information in a more direct format. One section there that will not be fully covered in this guide will be the [authentication APIs](/reference/api/authentication.md). There are a few options for authentication discussed there.
+
+For the sake of simplicity we can use the example of authentication using an API key. The key could be defined in the `@apostrophecms/express` module in a project like this:
+
+```javascript
+// modules/@apostrophecms/express/index.js
+module.exports = {
+  options: {
+    apiKeys: {
+      // Use your own strong, randomly generated key. Not like this one.
```

Comment:
Should we include a note about how these keys could also be stored as environment variables rather than in code? (Assuming that's possible...)

Reply:
Not as ENV variables, but they could be in data/local.js. That's covered more in the authentication page that I linked in the first paragraph in this section.

We should have more documentation of how to use environment configurations. I've created a ticket for that.